docs/source/programming.md

```diff
@@ -0,0 +1,123 @@
+
+:::{important}
+With Pulser, you program the interaction Hamiltonian by setting the distance between atoms, $R_{ij}$, and the choice of eigenstates used in the computation and the Rydberg level(s) targeted by the device. Most commonly, the ground and Rydberg states ($\left|g\right>$ and $\left|r\right>$) are used, such that
+$$
+H^\text{int}_{ij} = \frac{C_6}{R_{ij}^6} \hat{n}_i\hat{n}_j
+$$
+When providing the distance between atoms, Pulser ensures that you respect the constraints of your chosen device.
+:::
+
+#### 2.4. Evolution of the system's state
+
+For a system of atoms in the initial state $\left|\Psi_0\right>$, the final state of the system after a time $\Delta t$ is:
+$$ \left|\Psi_f\right> = \exp\left(-\frac{i}{\hbar}\int_0^{\Delta t} H(t) dt\right)$$
+
+## Writing a Pulser program
+
+As outlined above, Pulser lets you program an Hamiltonian so that you can manipulate the quantum state of a system of atoms, that encode the states. The series of necessary instructions is encapsulated in the so-called Pulser `Sequence`. Here is a step-by-step guide to create your own Pulser `Sequence`.
+
+### 1. Pick a Device
+
+```{mermaid}
+
+flowchart TB
+  A[Picking a Device] --> B{Do I want to run on a QPU}
+  B -->|Yes| C[Pick the Device from the QPU backend]
+  B -->|No| D[Do I want to be constrained by all the QPU specs ?]
+  D -->|Yes| E[Use a `Device` (like `pulser.AnalogDevice`)]
+  D --> |No| F[Use a `VirtualDevice` (like `pulser.MockDevice`)]
+
+```
+
+The `Device` you select will dictate some parameters and constrain others. For instance, the value of the $C_6$ and $C_3$ coefficients of the interaction Hamiltonian are defined by the device. For a complete view of the constraints introduced by the device, [check its description](./apidoc/core.rst).
+
+### 2. Create the Register
+
+The `Register` defines the position of the atoms. This determines:
+
+- the number of atoms to use in the quantum computation, i.e, the size of the system (let's note it $N$).
+- the distance between the atoms, the $R_{ij} (1\le i, j\le N)$ parameters in the interaction Hamiltonian.
+
+### 3. Pick the Channels
+
+A `Channel` targets the transition between two energy levels. Therefore, picking channels defines the energy levels that will be used in the computation. The channels must be picked from the `Device.channels`, so your device selection should take into account the channels it supports.
```

Comment:
a brief reference to the driving hamiltonian would make a good transition between pulser object and the hamiltonian presented above

Reply:
The fact is that the channel does not only refer to the drive hamiltonian but to both the interaction and the drive hamiltonian. This is what I explain in the next paragraph. And the association channel<-> drive hamiltonian takes me a whole paragraph to develop.

Reply:
Indeed...sorry I tried to look smart and comment in a rush